token-swap/program/src/curve/math.rs

```diff
@@ -201,6 +201,16 @@ impl PreciseNumber {
         Some(Self { value })
     }
 
+    /// Ceiling a precise value to a precision of ONE
+    pub fn ceiling(&self) -> Option<Self> {
+        let value = self
+            .value
+            .checked_add(one())?
+            .checked_div(one())?
+            .checked_mul(one())?;
+        Some(Self { value })
+    }
```

Comment:
Do you mind adding a test for `precise.ceiling() == precise.ceiling().ceiling()`? I think it will fail

Reply:
Will do, it definitely will fail

Reply:
Fixed in `95d7c7e`